sdk/messaging/azservicebus/receiver.go

```diff
@@ -59,6 +60,8 @@ type Receiver struct {
 
 	defaultDrainTimeout      time.Duration
 	defaultTimeAfterFirstMsg time.Duration
+
+	cancelReleaser *atomic.Value
```

Comment:
Why is this an `atomic.Value`?  Not saying it shouldn't be, but I'd like to understand the reasoning.

Reply:
Close() and Receive() don't have to happen in the same thread.

Reply:
I came to the same conclusion